krabs/krabs/ut.hpp

```diff
@@ -19,7 +19,13 @@ namespace krabs { namespace details {
     struct ut {
 
         typedef krabs::provider<> provider_type;
+        
+        struct filter_settings{
+            std::vector<unsigned short> provider_filter_event_ids_;
+            std::tuple<UCHAR, ULONGLONG, ULONGLONG, UCHAR> flags_tuple_;
+        };
 
+        typedef std::map<krabs::guid, filter_settings> provider_filter_settings;
```

Comment:
Should this be `map` or `unordered_map`? `unordered_map` is ultimately a Hash Map while `map` will be a BTree Map.

It doesn't look like order matters so I'd opt for `unordered_map`.